mongodump/metadata_dump.go

```diff
@@ -46,7 +46,7 @@ func (dump *MongoDump) dumpMetadata(intent *intents.Intent, buffer resettableOut
 	// The collection options were already gathered while building the list of intents.
 	// We convert them to JSON so that they can be written to the metadata json file as text.
 	if intent.Options != nil {
-		if meta.Options, err = bsonutil.ConvertBSONValueToJSON(intent.Options); err != nil {
+		if meta.Options, err = bsonutil.ConvertBSONValueToLegacyExtJSON(intent.Options); err != nil {
```

Comment:
I think we ought to convert to canonical Extended JSON -- I don't see a lot of value in using the legacy format for dumping metadata.  Anyone disagree?  A counterargument I would accept is that this ticket is all about getting rid of mgo/bson and we think we should make a change to a new format a separate ticket.   I guess I've talked myself into that point of view.  I'll open a ticket for it.

Reply:
hah, I had the exact same sequence of thoughts on this.